### PR TITLE
Add documentation for Event Counter task module

### PR DESCRIPTION
**Pull request by cornelinux**
Working on #1183
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23issuecomment-413139879%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221254%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221382%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221752%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221998%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210229947%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23issuecomment-413139879%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%23%20%5BCodecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Dh1%29%20Report%5Cn%3E%20Merging%20%5B%231184%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Ddesc%29%20into%20%5Bmaster%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/commit/7bf8b438817ad858f5bd1d57461ce346d8c8b202%3Fsrc%3Dpr%26el%3Ddesc%29%20will%20%2A%2Anot%20change%2A%2A%20coverage.%5Cn%3E%20The%20diff%20coverage%20is%20%60n/a%60.%5Cn%5Cn%5B%21%5BImpacted%20file%20tree%20graph%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184/graphs/tree.svg%3Fheight%3D150%26width%3D650%26token%3D7nHLZki60B%26src%3Dpr%29%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Dtree%29%5Cn%5Cn%60%60%60diff%5Cn%40%40%20%20%20%20%20%20%20%20%20%20%20Coverage%20Diff%20%20%20%20%20%20%20%20%20%20%20%40%40%5Cn%23%23%20%20%20%20%20%20%20%20%20%20%20master%20%20%20%20%231184%20%20%20%2B/-%20%20%20%23%23%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Coverage%20%20%2095.76%25%20%20%2095.76%25%20%20%20%20%20%20%20%20%20%20%20%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Files%20%20%20%20%20%20%20%20%20141%20%20%20%20%20%20141%20%20%20%20%20%20%20%20%20%20%20%5Cn%20%20Lines%20%20%20%20%20%20%2017112%20%20%20%2017112%20%20%20%20%20%20%20%20%20%20%20%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Hits%20%20%20%20%20%20%20%2016388%20%20%20%2016388%20%20%20%20%20%20%20%20%20%20%20%5Cn%20%20Misses%20%20%20%20%20%20%20%20724%20%20%20%20%20%20724%5Cn%60%60%60%5Cn%5Cn%5Cn%5Cn------%5Cn%5Cn%5BContinue%20to%20review%20full%20report%20at%20Codecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Dcontinue%29.%5Cn%3E%20%2A%2ALegend%2A%2A%20-%20%5BClick%20here%20to%20learn%20more%5D%28https%3A//docs.codecov.io/docs/codecov-delta%29%5Cn%3E%20%60%5Cu0394%20%3D%20absolute%20%3Crelative%3E%20%28impact%29%60%2C%20%60%5Cu00f8%20%3D%20not%20affected%60%2C%20%60%3F%20%3D%20missing%20data%60%5Cn%3E%20Powered%20by%20%5BCodecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Dfooter%29.%20Last%20update%20%5B7bf8b43...4d359ae%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1184%3Fsrc%3Dpr%26el%3Dlastupdated%29.%20Read%20the%20%5Bcomment%20docs%5D%28https%3A//docs.codecov.io/docs/pull-request-comments%29.%5Cn%22%2C%20%22created_at%22%3A%20%222018-08-15T09%3A11%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/in/254%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/apps/codecov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%204d359aea5dd4f780dbc374794958066de619fb30%20doc/periodictask/eventcounter.rst%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221998%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22...that%20display%20%2A%2As%2A%2A%20the%20absolute...%22%2C%20%22created_at%22%3A%20%222018-08-15T09%3A54%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/periodictask/eventcounter.rst%3AL3-42%22%7D%2C%20%22Pull%204d359aea5dd4f780dbc374794958066de619fb30%20doc/periodictask/eventcounter.rst%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221254%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Using%20the%20Event%20Counter%20_%20%2A%2A%2C%2A%2A%20_%20privacyIDEA...%22%2C%20%22created_at%22%3A%20%222018-08-15T09%3A51%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/periodictask/eventcounter.rst%3AL3-42%22%7D%2C%20%22Pull%204d359aea5dd4f780dbc374794958066de619fb30%20doc/periodictask/eventcounter.rst%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221752%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22The%20event%20counter%20key%20store%20%2A%2As%2A%2A%20the%20current%20number...%22%2C%20%22created_at%22%3A%20%222018-08-15T09%3A53%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/periodictask/eventcounter.rst%3AL3-42%22%7D%2C%20%22Pull%204d359aea5dd4f780dbc374794958066de619fb30%20doc/periodictask/eventcounter.rst%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1184%23discussion_r210221382%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22...current%20value%20%2A%2Aas%20a%2A%2A%20time%20series...%22%2C%20%22created_at%22%3A%20%222018-08-15T09%3A52%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20current%20value%20is%20not%20written%20as%20a%20time%20series.%5Cr%5CnI%20think%20the%20current%20value%20is%20added%20to%20a%20time%20series%3F%20%28as%20the%20time%20series%20consists%20of%20a%20lot%20of%20current%20values%29%3F%22%2C%20%22created_at%22%3A%20%222018-08-15T10%3A35%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/periodictask/eventcounter.rst%3AL3-42%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1184#issuecomment-413139879'>General Comment</a></b>
- <a href='https://github.com/apps/codecov'></a> # [Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=h1) Report
> Merging [#1184](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=desc) into [master](https://codecov.io/gh/privacyidea/privacyidea/commit/7bf8b438817ad858f5bd1d57461ce346d8c8b202?src=pr&el=desc) will **not change** coverage.
> The diff coverage is `n/a`.
[Impacted file tree graph](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=tree)
```diff
@@           Coverage Diff           @@
##           master    #1184   +/-   ##
=======================================
Coverage   95.76%   95.76%
=======================================
Files         141      141
Lines       17112    17112
=======================================
Hits        16388    16388
Misses        724      724
```
------
[Continue to review full report at Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=footer). Last update [7bf8b43...4d359ae](https://codecov.io/gh/privacyidea/privacyidea/pull/1184?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).
- [ ] <a href='#crh-comment-Pull 4d359aea5dd4f780dbc374794958066de619fb30 doc/periodictask/eventcounter.rst 7'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1184#discussion_r210221254'>File: doc/periodictask/eventcounter.rst:L3-42</a></b>
- <a href='https://github.com/plettich'></a> Using the Event Counter _ **,** _ privacyIDEA...
- [ ] <a href='#crh-comment-Pull 4d359aea5dd4f780dbc374794958066de619fb30 doc/periodictask/eventcounter.rst 11'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1184#discussion_r210221382'>File: doc/periodictask/eventcounter.rst:L3-42</a></b>
- <a href='https://github.com/plettich'></a> ...current value **as a** time series...
- <a href='https://github.com/cornelinux'></a> The current value is not written as a time series.
I think the current value is added to a time series? (as the time series consists of a lot of current values)?
- [ ] <a href='#crh-comment-Pull 4d359aea5dd4f780dbc374794958066de619fb30 doc/periodictask/eventcounter.rst 29'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1184#discussion_r210221752'>File: doc/periodictask/eventcounter.rst:L3-42</a></b>
- <a href='https://github.com/plettich'></a> The event counter key store **s** the current number...
- [ ] <a href='#crh-comment-Pull 4d359aea5dd4f780dbc374794958066de619fb30 doc/periodictask/eventcounter.rst 40'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1184#discussion_r210221998'>File: doc/periodictask/eventcounter.rst:L3-42</a></b>
- <a href='https://github.com/plettich'></a> ...that display **s** the absolute...

<a href='https://www.codereviewhub.com/privacyidea/privacyidea/pull/1184?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/privacyidea/privacyidea/pull/1184?approve=1'></a>&nbsp;<a href='https://github.com/privacyidea/privacyidea/pull/1184'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by cornelinux** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/privacyidea/privacyidea/pull/1184'>Refresh</a>.<a href='#crh-update-none'></a>

File: doc/periodictask/eventcounter.rst [hunk 1]
```diff
@@ -3,4 +3,39 @@
 EventCounter
 ------------
 
-TODO
\ No newline at end of file
+The Event Counter task module can be used with the :ref:`periodic_tasks` to create time series of certain events.
+An event could be a failed authentication request. Using the Event Counter privacyIDEA can create graphs that display
```
Comment (plettich) [inline doc/periodictask/eventcounter.rst]: Using the Event Counter _ **,** _ privacyIDEA...

File: doc/periodictask/eventcounter.rst [hunk 2]
```diff
@@ -3,4 +3,39 @@
...
+the development of failed authentication requests over time.
+
+To do this, the Event Counter task module reads a counter value from the database table ``EventCounter`` and writes this
+current value in the time series in the database table ``MonitoringStats``.
```
Comment (plettich) [inline doc/periodictask/eventcounter.rst]: ...current value **as a** time series...

File: doc/periodictask/eventcounter.rst [hunk 3]
```diff
@@ -3,4 +3,39 @@
...
+
+**event_counter**
+
+    This is the name of the event counter key, that was defined in a :ref:`counterhandler` definition and that is
+    read from the database table ``EventCounter``.
+
+**stats_key**
+
+    This is the name of the statistics key that is written to the ``MonitoringStats`` database table.
+    The event counter key store the current number of counted events, the ``stats_key`` takes the current number
```
Comment (plettich) [inline doc/periodictask/eventcounter.rst]: The event counter key store **s** the current number...

File: doc/periodictask/eventcounter.rst [hunk 4]
```diff
@@ -3,4 +3,39 @@
...
+
+**reset_event_counter**
+
+    This is a boolean value. If it is set to true (the checkbox is checked), then the event counter will be reset to zero,
+    after the task module has read the key.
+
+    Resetting the the event counter results in a time series of "events per time interval". The time intervall is
+    specified by the time intervall in which the Event Counter task module is called.
+    If ``reset_event_counter`` is not checked, then the event handler will continue to increase the counter value.
+    Use this, if you want to create a time series, that display the absolute number of events.
```
Comment (plettich) [inline doc/periodictask/eventcounter.rst]: ...that display **s** the absolute...

File: doc/periodictask/eventcounter.rst (hunk 2 above)
Comment (cornelinux) [reply doc/periodictask/eventcounter.rst]: The current value is not written as a time series.
I think the current value is added to a time series? (as the time series consists of a lot of current values)?

**Pull request merged**